TrafficCapture/replayerPlugins/jsonMessageTransformers/Es2OsPost23/src/main/java/org/opensearch/migrations/transform/Es2OsPost23.java

```diff
@@ -1,9 +1,30 @@
 package org.opensearch.migrations.transform;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
 public class Es2OsPost23 implements IJsonTransformerProvider {
 
+    @Override
     public JsonJMESPathTransformer createTransformer(Object jsonConfig) {
-        String script = "{ \"settings\": settings, \"mappings\": payload.mappings.*.properties | [merge(@)] }";
+        ObjectMapper objectMapper = new ObjectMapper();
+        Map<String, Object> jsonMap = objectMapper.convertValue(jsonConfig, Map.class);
+        String script;
+
+        if (jsonMap.containsKey("mappings") && jsonMap.get("mappings") instanceof Map) {
```

Comment:
the config value coming in is meant to configure a transformer (like how you're passing the script to the JMESPath Transformer's c'tor).  TransformationLoader only creates ONE transformer from a provider AND it passes the configuration, NOT a document, to createTransformer().

You'll want to make sure that you add another test that uses TransformationLoader to configure this transformer and then give it a couple documents.  Those tests would have failed & you could have spotted the inconsistencies.